userspace/libsinsp/sinsp.cpp

```diff
@@ -1686,6 +1686,13 @@ void sinsp::set_filter(const std::string& filter)
 	m_internal_flt_ast = compiler.get_filter_ast();
 }
 
+void sinsp::set_filter(const std::string& filterstring, std::unique_ptr<sinsp_filter> filter)
```

Comment:
What about adding `filterstring` as optional parameter to existent `set_filter` instead of a brand new method?